.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,20 @@
+# https://pre-commit.com
+# This GitHub Action assumes that the repo contains a valid .pre-commit-config.yaml file.
+---
+name: pre-commit
+on: [push]
+
+permissions:
+  contents: read
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.11'
```

Comment:
unlikely to be an issue but the codebase requires >=3.10 so maybe we should use that here.

Reply:
Changed to use Python 3.10